core/art/views.py

```diff
@@ -1131,14 +1144,47 @@ def registerARTTest(request):
     tarindex = None
     inputfileid = None
     gitlabid = None
+    test_type = None
 
     # log all the req params for debug
     _logger.debug('[ART] registerARTtest requestParams: ' + str(request.session['requestParams']))
 
     # Checking whether params were provided
+    if 'test_type' in request.session['requestParams']:
+        test_type = request.session['requestParams']['test_type']
+
     if 'requestParams' in request.session and 'pandaid' in request.session['requestParams'] and 'testname' in request.session['requestParams']:
         pandaid = request.session['requestParams']['pandaid']
         testname = request.session['requestParams']['testname']
+    elif test_type == 'local':
+        '''
+        session_id = request.COOKIES.get('sessionid')
+        if session_id:
+        '''
+        client_ip = get_client_ip(request)
+        try:
+        # Perform reverse DNS lookup to get the client's hostname
+            art_host = socket.gethostbyaddr(client_ip)[0]
+        except socket.herror:
+        # If reverse DNS lookup fails, return the IP address as fallback
+            art_host = client_ip
+
+        if art_host.split('.')[0] in art_const.AUTHORIZED_HOSTS: 
+            pass
+        else:
+            return JsonResponse({"error": "Invalid ART API user!"}, status=403)
+
+        # Generate job ID for ART Local
+        query = {'test_type': 'local'}
+        localIDs = []
+        localIDs.extend(ARTTests.objects.filter(**query).values('pandaid'))
+        if localIDs:
+            newIDs = sorted(localIDs, key=lambda x: x['pandaid'])
+            pandaid = int(newIDs[-1]['pandaid']) + 1
+        else:
+            pandaid = art_const.INITIAL_LOCAL_ID
+        _logger.info("JobID: {} was generated".format(pandaid))
+        testname = request.session['requestParams']['testname']
```

Comment:
this line is the same for both local and grid jobs, and it is duplicated with 1158, it is better to move it out of if statement.

Reply:
Now both local and grid have the commond code.